src/main.ts

```diff
@@ -124,26 +125,47 @@ export default class VaultChat extends Plugin {
 	onunload() {
 	}
 
-	async searchForTerm(searchTerm: string): Promise<Array<string>> {
-		if (searchTerm === '') {
-			return []
-		}
-		const embedding = await this.openAIHandler.createEmbedding(searchTerm)
-		if (embedding === undefined) {
-			console.error(`Failed to generate vector for search term.`)
-			return []
+	async loadSettings() {
+		this.settings = Object.assign({}, DEFAULT_SETTINGS, await this.loadData());
+	}
+
+	async saveSettings() {
+		await this.saveData(this.settings);
+		if (this.waitingForApiKey && this.apiKeyIsValid()) {
+			this.waitingForApiKey = false
+			await this.initializePlugin()
 		}
-		const nearest: NearestVectorResult[] = this.vectorStore.getNearestVectors(embedding, 3, this.settings.relevanceThreshold)
-		return nearest.map(n => n.path)
 	}
 
-	async getSearchResultsFiles(searchTerm: string): Promise<Array<SearchResult>> {
-		const embeddingResponse = await this.openAIHandler.createEmbedding(searchTerm)
-		if (embeddingResponse === undefined) {
-			console.error(`Failed to generate vector for search term.`)
-			return []
+	async askChatGpt(question: string) {
+		// HyDE: request note that answers the question
+		const conversation: Array<ChatCompletionRequestMessage> = []
+		const hydeMessage: ChatCompletionRequestMessage = {
+			role: ChatCompletionResponseMessageRoleEnum.User,
+			content: `${hydePrompt} ${question}`,
 		}
-		const nearestVectors = this.vectorStore.getNearestVectors(embeddingResponse, 8, this.settings.relevanceThreshold)
+		conversation.push(hydeMessage)
+		const hydeResponse = await this.openAIHandler.createChatCompletion(conversation)
+		if (!hydeResponse || hydeResponse.choices.length === 0 || !hydeResponse.choices[0].message) {
+			console.error(`Failed to get hyde response for query: ${question}`)
+			return
+		}
+
+		// create embeddings for that note and all of its blocks
+		const hydeNote = hydeResponse.choices[0].message.content
+		const hydeNoteBlocks = parseMarkdown(hydeNote, '')
+		const hydeBlocksStrings = hydeNoteBlocks.map(t => `${t.path} ${t.localHeading} ${t.content}`)
```

Comment:
nit: name on this is a little misleading since it holds mutliple blocs, maybe?
```suggestion
		const queryBlockStrings = hydeNoteBlocks.map(t => `${t.path} ${t.localHeading} ${t.content}`)
```